src/device/clint.rs

```diff
@@ -95,12 +168,39 @@ impl VirtClint {
         }
     }
 
-    pub fn write_clint(
+    pub fn write_clint_from_payload(
+        &self,
+        ctx: &mut VirtContext,
+        mctx: &mut MiralisContext,
+        value: usize,
+    ) {
+        self.write_clint(
+            MTIMECMP_OFFSET + 8 * mctx.hw.hart,
+            Width::Byte8,
+            value,
+            ctx,
+            true,
+        )
+        .unwrap();
+    }
+
+    pub fn write_clint_from_firmwarwe(
+        &self,
+        offset: usize,
+        w_width: Width,
+        value: usize,
+        ctx: &mut VirtContext,
+    ) -> Result<(), &'static str> {
+        self.write_clint(offset, w_width, value, ctx, false)
+    }
+
+    fn write_clint(
         &self,
         offset: usize,
         w_width: Width,
         value: usize,
         ctx: &mut VirtContext,
+        is_from_payload: bool,
```

Comment:
I would suggest creating a small enum, something like `ClintTimer::Payload` and `ClintTimer::Firmware`. It would be a bit more readable, but mostly I think we might eventually add a third one for Miralis itself (for instance if we want a policy that guarantees an enclave is going to run for X ms).